### ISSUE TITLE
PHP Fatal error:  Uncaught TypeError: Cannot assign Composer\PartialComposer to property NunoMaduro\PhpInsights\Domain\ComposerLoader::$composer of type ?Composer\Composer

### ISSUE THREAD

**Issue by vaawebdev** [nunomaduro/phpinsights#565]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | v2.1.0

'preset' => 'laravel',

Running `./vendor/bin/phpinsights` results an error

Locked composer version "dev-main" 

```
PHP Fatal error:  Uncaught TypeError: Cannot assign Composer\PartialComposer to property NunoMaduro\PhpInsights\Domain\ComposerLoader::$composer of type ?Composer\Composer in /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Domain/ComposerLoader.php:30
Stack trace:
#0 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Domain/Insights/Composer/ComposerLockMustBeFresh.php(22): NunoMaduro\PhpInsights\Domain\ComposerLoader::getInstance()
#1 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Domain/Results.php(155): NunoMaduro\PhpInsights\Domain\Insights\Composer\ComposerLockMustBeFresh->hasIssue()
#2 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Domain/Results.php(66): NunoMaduro\PhpInsights\Domain\Results->getPercentage()
#3 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Application/Console/Formatters/Console.php(253): NunoMaduro\PhpInsights\Domain\Results->getStructure()#4 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Application/Console/Formatters/Console.php(135): NunoMaduro\PhpInsights\Application\Console\Formatters\Console->summary()
#5 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Application/Console/Formatters/Multiple.php(32): NunoMaduro\PhpInsights\Application\Console\Formatters\Console->format()
#6 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Application/Console/Analyser.php(38): NunoMaduro\PhpInsights\Application\Console\Formatters\Multiple->format()
#7 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Application/Console/Commands/AnalyseCommand.php(65): NunoMaduro\PhpInsights\Application\Console\Analyser->analyse()
#8 [internal function]: NunoMaduro\PhpInsights\Application\Console\Commands\AnalyseCommand->__invoke()
#9 /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Application/Console/Commands/InvokableCommand.php(41): call_user_func()
#10 /var/www/workflowz/vendor/symfony/console/Command/Command.php(291): NunoMaduro\PhpInsights\Application\Console\Commands\InvokableCommand->execute()       
#11 /var/www/workflowz/vendor/symfony/console/Application.php(989): Symfony\Component\Console\Command\Command->run()
#12 /var/www/workflowz/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#13 /var/www/workflowz/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun()
#14 /var/www/workflowz/vendor/nunomaduro/phpinsights/bin/phpinsights(42): Symfony\Component\Console\Application->run()
#15 /var/www/workflowz/vendor/nunomaduro/phpinsights/bin/phpinsights(43): {closure}()
#16 /var/www/workflowz/vendor/bin/phpinsights(107): include('...')
#17 {main}
  thrown in /var/www/workflowz/vendor/nunomaduro/phpinsights/src/Domain/ComposerLoader.php on line 30
```

**Comment by catabozan** [comment]
I'm getting the same error, just installed the library.

PHP: 8.1.3
Laravel: 9.2.0
phpinsights: 2.0.1

**Comment by shomsy** [comment]
I have same error:

![image](https://user-images.githubusercontent.com/3872827/155453661-4a42c950-e035-45bb-aba5-a7a7e73025cb.png)

**Comment by 50bhan** [comment]
The root of the issue is the composer. As the composer maintainer said in #559, we need to remove the `composer/composer` dependency. Because they split the class into a base class `PartialComposer` (https://github.com/composer/composer/pull/10547).

@nunomaduro @Jibbarth @olivernybroe Could you guys please help with this? As it's not possible to install the package anymore.

**Comment by nunomaduro** [comment]
The people reporting this issue, are free to make a pull request. Personally, I don't have time for this project. 👍🏻

**Comment by Queopius** [comment]
Hello everyone! I have the same problem.
![Screenshot from 2022-02-25 10-48-23](https://user-images.githubusercontent.com/10995315/155693868-639fae72-f4e3-4a00-9c49-35e4011bf0d4.png)

**Comment by damek24** [comment]
temp fix that worked for me:
in `src\Domain\ComposerLoader.php` change `fullLoad` parameter to true:
`self::$composer = (new Factory())->createComposer($io, ComposerFinder::getPath($collector), false, null, true);`

**Comment by DannyvdSluijs** [comment]
> temp fix that worked for me:
> in `src\Domain\ComposerLoader.php` change `fullLoad` parameter to true:
> `self::$composer = (new Factory())->createComposer($io, ComposerFinder::getPath($collector), false, null, true);`

That is what the pr in #566 does.

**Comment by cmgmyr** [comment]
fixed in #566

**Comment by Sergiobop** [comment]
Can we have a release with the fix @cmgmyr ? Thanks!
